service/src/main.rs

```diff
@@ -338,16 +339,31 @@ fn start_worker<E, T, D>(
 	let tx_hash = node_api.send_extrinsic(xthex, XtStatus::Finalized).unwrap();
 	println!("[<] Extrinsic got finalized. Hash: {:?}\n", tx_hash);
 
-	let latest_head = init_light_client(&node_api, enclave.as_ref());
+	let last_synced_header = init_light_client(&node_api, enclave.as_ref());
 	println!("*** [+] Finished syncing light client\n");
 
 	// ------------------------------------------------------------------------
 	// start interval block production
-	let api4 = node_api.clone();
+	let side_chain_enclave_api = enclave.clone();
 	thread::Builder::new()
 		.name("interval_block_production_timer".to_owned())
+		.spawn(move || start_interval_block_production(side_chain_enclave_api.as_ref()))
+		.unwrap();
+
+	// ------------------------------------------------------------------------
+	// start parent chain syncing loop (subscribe to header updates)
+	let api4 = node_api.clone();
+	thread::Builder::new()
+		.name("parent_chain_sync_loop".to_owned())
 		.spawn(move || {
-			start_interval_block_production(enclave.clone().as_ref(), &api4, latest_head)
+			if let Err(e) = subscribe_to_parentchain_new_headers(
+				enclave.clone().as_ref(),
+				&api4,
+				last_synced_header,
+			) {
+				error!("Parent chain block syncing terminated with a failure: {:?}", e);
```

Comment:
I think we agreed once to write it as one one: `Parentchain` (also visible in the variable naming).

Reply:
Regardless, that's probably a discussion for another time 😉 I'll change the messages to 'parentchain' 👍